service/src/test/java/bio/terra/workspace/service/workspace/flight/azure/ValidateLandingZoneAgainstPolicyStepTest.java

```diff
@@ -0,0 +1,170 @@
+    var workspacePao = createPao();
+    when(tpsApiDispatch.getOrCreatePao(workspaceId, TpsComponent.WSM, TpsObjectType.WORKSPACE))
+        .thenReturn(workspacePao);
+    assertThrows(RegionNotAllowedException.class, () -> step.doStep(mock()));
+  }
+
+  @Test
+  void doStep_validatesProtectedDataLandingZone() throws Exception {
+    var landingZoneId = UUID.randomUUID();
+    Workspace workspace = mock();
+
+    when(workspaceService.getWorkspace(workspaceId)).thenReturn(workspace);
+    when(landingZoneApiDispatch.getLandingZoneId(new BearerToken(token), workspace))
+        .thenReturn(landingZoneId);
+    when(landingZoneApiDispatch.getLandingZoneRegionUsingWsmToken(landingZoneId))
+        .thenReturn("test-region");
+    when(tpsApiDispatch.listValidRegions(workspaceId, CloudPlatform.AZURE))
+        .thenReturn(Arrays.asList("test-region"));
+    var workspacePao =
+        createPao(
+            new TpsPolicyInput()
+                .namespace(TpsUtilities.TERRA_NAMESPACE)
+                .name(TpsUtilities.PROTECTED_DATA_POLICY_NAME));
+    when(tpsApiDispatch.getOrCreatePao(workspaceId, TpsComponent.WSM, TpsObjectType.WORKSPACE))
+        .thenReturn(workspacePao);
+    when(tpsApiDispatch.getPao(workspaceId)).thenReturn(workspacePao);
+    ApiAzureLandingZone landingZone = mock();
+    when(landingZoneApiDispatch.getAzureLandingZone(new BearerToken(token), landingZoneId))
+        .thenReturn(landingZone);
+    when(policyValidator.validateLandingZoneSupportsProtectedData(landingZone))
+        .thenReturn(List.of());
+
+    var result = step.doStep(mock());
+    assertEquals(StepResult.getStepResultSuccess(), result);
+  }
+
+  @Test
+  void doStep_failsOnLZIncompatibleWithPolicy() throws Exception {
+    var landingZoneId = UUID.randomUUID();
+    Workspace workspace = mock();
+    when(workspaceService.getWorkspace(workspaceId)).thenReturn(workspace);
+    when(landingZoneApiDispatch.getLandingZoneId(new BearerToken(token), workspace))
+        .thenReturn(landingZoneId);
+    when(landingZoneApiDispatch.getLandingZoneRegionUsingWsmToken(landingZoneId))
+        .thenReturn("test-region");
+    when(tpsApiDispatch.listValidRegions(workspaceId, CloudPlatform.AZURE))
+        .thenReturn(Arrays.asList("test-region"));
+    var workspacePao =
+        createPao(
+            new TpsPolicyInput()
+                .namespace(TpsUtilities.TERRA_NAMESPACE)
+                .name(TpsUtilities.PROTECTED_DATA_POLICY_NAME));
+    when(tpsApiDispatch.getOrCreatePao(workspaceId, TpsComponent.WSM, TpsObjectType.WORKSPACE))
+        .thenReturn(workspacePao);
+    when(tpsApiDispatch.getPao(workspaceId)).thenReturn(workspacePao);
+    ApiAzureLandingZone landingZone = mock();
+    when(landingZoneApiDispatch.getAzureLandingZone(new BearerToken(token), landingZoneId))
+        .thenReturn(landingZone);
+    var validationErrors = List.of("invalid policy");
+    when(policyValidator.validateLandingZoneSupportsProtectedData(landingZone))
```

Comment:
The actual validation calls are being mocked here, because they are tested separately, allowing the step tests to focus on ensuring the validation is called correctly, and errors are handled appropriately.